exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetScanBatchCreator.java

```diff
@@ -107,7 +107,7 @@ public ScanBatch getBatch(FragmentContext context, ParquetRowGroupScan rowGroupS
         if (!context.getOptions().getOption(ExecConstants.PARQUET_NEW_RECORD_READER).bool_val && !isComplex(footers.get(e.getPath()))) {
           readers.add(
               new ParquetRecordReader(
-                  context, e.getPath(), e.getRowGroupIndex(), fs,
+                  context, rowGroupScan.getBatchSize(), e.getPath(), e.getRowGroupIndex(), fs,
```

Comment:
If it's only for one type of parquet reader, please document it in the JIRA, so that people will know this.